File: Model-View-ViewModel-Java/app/src/main/java/com/mvvm/java/core/base/BaseViewHolder.java
package com.mvvm.java.core.base;

import androidx.recyclerview.widget.RecyclerView;
import android.view.View;

public abstract class BaseViewHolder extends RecyclerView.ViewHolder {

    private int currentPosition;

    public BaseViewHolder(View itemView) {
        super(itemView);
    }

    protected abstract void clear();

    public void onBind(int position) {
        currentPosition = position;
        clear();
    }

    public int getCurrentPosition() {
        return currentPosition;
    }
}

File: Model-View-ViewModel-Kotlin/app/src/main/java/com/base/mvvm/core/base/BaseViewHolder.kt
package com.base.mvvm.core.base

import android.view.View

import androidx.recyclerview.widget.RecyclerView

abstract class BaseViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    private var currentPosition = 0

    open fun onBind(position: Int) {
        this.currentPosition = position
        clear()
    }

    protected abstract fun clear()

    fun getCurrentPosition(): Int {
        return this.currentPosition
    }
}